### PR TITLE
Eliminate as much usize as making sense from CKB VM

### PR DESCRIPTION
**Pull request by xxuejie**
`usize` is a tricky part since it has different widths across 32-bit vs 64-bit architectures. This PR tries to remove the use of usize from the parts that are making sense. After this PR, we should only be using usize where we have to(such as indexing arrays), and constants where we know no overflows would occur.

### CODE REVIEW

**Comment by xxuejie** [comment]
The failed CI check is due to codecov mis-configuration. I've revised the configuration, and it should work all right in future PRs.

File: src/machine/asm/mod.rs [hunk 1]
```diff
@@ -76,110 +78,121 @@ impl Memory<u64> for Box<AsmCoreMachine> {
...
-    fn execute_load16(&mut self, addr: usize) -> Result<u16, Error> {
+    fn execute_load16(&mut self, addr: u64) -> Result<u16, Error> {
         check_permission(self, addr, 2, FLAG_EXECUTABLE)?;
         self.load16(&(addr as u64)).map(|v| v as u16)
     }
 
     fn load8(&mut self, addr: &u64) -> Result<u64, Error> {
-        let addr = *addr as usize;
-        if addr + 1 > self.memory.len() {
+        let addr = *addr as u64;
```
Comment (doitian) [inline src/machine/asm/mod.rs]: Many redudant `as u64` in this file.

Comment (xxuejie) [reply src/machine/asm/mod.rs]: Ooops I was relying too much on clippy, already resolved.

File: src/jit/machine.rs [hunk 2]
```diff
@@ -284,7 +284,7 @@ impl BaselineJitMachine {
                 } else {
                     // TODO: check if dynasm handles alignments here.
                     emitter.encode(&mut buffer_mut[rust_data.used..])?;
-                    let offset = rust_data.used;
+                    let offset = rust_data.used as u64;
```
Comment (jjyr) [inline src/jit/machine.rs]: I notice the `used` field is still usize type, not sure if this is missed or is intentioned.

Comment (xxuejie) [reply src/jit/machine.rs]: Yes this is kept as usize for 2 reasons:

1. Actually [here](https://github.com/nervosnetwork/ckb-vm/blob/2ffd6f12b750949c88103f26eb2b27e762ff1d8f/src/jit/machine.rs#L281) we ensure that used can never exceed 1MB, which will fit well within usize even for 32-bit platform
2. The whole JIT package right now is still treated as a PoC and won't be included in production, in fact, I'm thinking of removing the whole JIT package once the new AOT mode is in place, that's why this PR only contains necessary fixes in the JIT package, and still many fields in JIT package as usize.

**Pull request merged**